### PR TITLE
Fix timezone fallback in ingest processor

### PR DESCRIPTION
**Pull request by spinscale**
If no timezone was specified in the date processor, then the conversion
would lead to wrong time, as UTC was assumed by default, leading to
wrongly stored dates.

This commit does not assume a default timezone and will thus not format
the dates in a wrong way.

Unfortunately we never had any tests testing this, but this was found via the beats integration tests.

### CODE REVIEW

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java [hunk 1]
```diff
@@ -88,9 +91,12 @@ private long parseMillis(String date) {
             }
 
             int year = LocalDate.now(ZoneOffset.UTC).getYear();
-            DateFormatter formatter = DateFormatter.forPattern(format)
-                .withLocale(locale)
-                .withZone(zoneId);
+            DateFormatter dateFormatter = DateFormatter.forPattern(format)
+                .withLocale(locale);
```
Comment (pgomulka) [inline modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java]: I suppose locale cannot be null? 
the if condition in JavaDateFormatter is reversed compared to JodaDateFormatter which might throw NPE?

**Review by pgomulka — APPROVED** [review]
It looks good to me. I think it is unlikely that Locale would be null (can't see that possibility in code)

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java (hunk 1 above)
Comment (spinscale) [reply modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java]: indeed, cannot be null, fallback is done in the date processor configuration

**Comment by spinscale** [comment]
@pgomulka can you take another look? I looked at the diff and figured I had reimplemented the earlier null check in the processors in the date format class. All that was needed were a few changes in the DateFormat.Java part of things... sorry for the iteration, but the code is much shorter now

**Review by pgomulka — APPROVED** [review]
nice - I like those extra test cases
LGTM

**Pull request merged**